src/main/java/org/zalando/nakadi/validation/EventTypeOptionsValidator.java

```diff
@@ -28,13 +30,16 @@ public void validate(final Object target, final Errors errors) {
     }
 
     private void checkRetentionTime(final Errors errors, final EventTypeOptions options) {
-        if (Objects.nonNull(options) && Objects.nonNull(options.getRetentionTime())) {
-            final Long retentionTime = options.getRetentionTime();
-            if (retentionTime > maxTopicRetentionMs) {
-                createError(errors, "can not be more than " + maxTopicRetentionMs);
-            } else if (retentionTime < minTopicRetentionMs) {
-                createError(errors, "can not be less than " + minTopicRetentionMs);
-            }
+        final Long retentionTime = options.getRetentionTime();
+        if (retentionTime == null) {
+            options.setRetentionTime(defaultTopicRetentionMs);
```

Comment:
For me it looks very strange that validator updates value. I think Validator should only validate and nothing more.